rest-client/src/main/java/io/apicurio/registry/client/RegistryRestClientFactory.java

```diff
@@ -20,34 +20,23 @@
 import io.apicurio.registry.auth.Auth;
 import okhttp3.OkHttpClient;
 
-import java.util.HashMap;
 import java.util.Map;
 
-import static io.apicurio.registry.client.request.RestClientConfig.REGISTRY_REQUEST_HEADER_AUTHORIZATION;
-
 
 /**
  * @author eric.wittmann@gmail.com
  */
 public class RegistryRestClientFactory {
 
-
     public static RegistryRestClient create(String baseUrl) {
         return new RegistryRestClientImpl(baseUrl);
     }
 
-    public static RegistryRestClient create(String baseUrl, Auth auth) {
-        final Map<String, Object> config = new HashMap<>();
-        //TODO carnalca for now we only support bearer auth here
-        config.put(REGISTRY_REQUEST_HEADER_AUTHORIZATION, auth.getAuthStrategy().getToken());
-        return new RegistryRestClientImpl(baseUrl, config);
-    }
-
     public static RegistryRestClient create(String baseUrl, OkHttpClient okHttpClient) {
         return new RegistryRestClientImpl(baseUrl, okHttpClient);
     }
 
-    public static RegistryRestClient create(String baseUrl, Map<String, Object> configs) {
-        return new RegistryRestClientImpl(baseUrl, configs);
+    public static RegistryRestClient create(String baseUrl, Map<String, Object> configs, Auth auth) {
+        return new RegistryRestClientImpl(baseUrl, configs, auth);
```

Comment:
Let's have another variant of the `create()` method that takes the `Auth` instance instead of changing the existing one.  That way you don't have to pass `null` when you don't have an `Auth`.